include/matx/core/tensor_utils.h

```diff
@@ -509,15 +510,23 @@ namespace detail {
   {
     MATX_NVTX_START("", matx::MATX_NVTX_LOG_API)
 
+    using namespace std::literals::string_literals;
+
     if constexpr (is_complex_v<T>) {
-      printf("%.4e%+.4ej ", static_cast<float>(val.real()),
+      auto prec = std::to_string(PRINT_PRECISION);
```

Comment:
Should these be `const auto`?

Reply:
Thanks for your reply and help~ I pushed two additional commits to this branch according to your recommandations, can you check whether it is OK now?